Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -439,6 +479,9 @@ void DirectoryBlobReader::SetNonpartitionDiscHeader(const std::vector<u8>& parti
   if (header_bin_bytes_read < 0x61)
     m_disc_header_nonpartition[0x61] = 0;
 
+  m_encrypted = std::all_of(m_disc_header_nonpartition.data() + 0x60,
```

Comment:
This looks like a very complicated way to check if one u32 value equals zero. Is there a struct for this header?

Reply:
> This looks like a very complicated way to check if one u32 value equals zero.

Hmm... Would it be safe to use `reinterpret_cast<u32*>` here? I don't know if the start of the vector is guaranteed to be correctly aligned.

> Is there a struct for this header?

No.

> Can you add a small comment why this flag is required?

Well, it's required in order to be handle both unencrypted and encrypted discs, and I'm not how useful a comment that just says something like that would be. The full details are available at http://wiibrew.org/wiki/Wii_Disc.